File: runtimecs/java/util/Collection.cs
using java.lang;

namespace java.util 
{ 
    public interface Collection: Iterable
    {   
        bool add(object o);
        bool addAll(Collection c);
        void clear(); 
        bool contains(object o);
        bool containsAll(Collection c);
        bool Equals(object o);
        bool remove(object o);
        bool removeIf(java.util.function.Predicate predicate);
        bool removeAll(Collection c);
        bool retainAll(Collection c);
        int GetHashCode();
        bool isEmpty();
    //  Iterator iterator();
        int size();
        object[] toArray();
        object[] toArray(object[]a);
        string ToString();
    }

    public static class Collection_0009
    {
        public static bool removeIf(Collection @this, java.util.function.Predicate predicate)
        {   
            java.util.Iterator i = @this.iterator();
            bool didremove = false;
            while (i.hasNext()) 
            {   
                object o = i.next();
                if (predicate.test(o)) 
                {   
                    didremove = true;
                    i.remove();
                }
            }
            return didremove;
        }
    }
}

File: runtimejs/java/util/Collection.js
//load// java/lang/Iterable
var java_util_Collection = { 
    _superinterfaces: [java_lang_Iterable], 
    _defaults:
    {   removeIf_1: function(predicate)
        {   var i = this.iterator_0();
            while (i.hasNext_0()) 
            {   var o = i.next_0();
                if (predicate.test_1(o)) i.remove_0();
            }
        }
        ,
    }

}; 

// -- methods:
// boolean	contains(Object o)
// boolean	containsAll(Collection<?> c)
// boolean	equals(Object o)
// int	hashCode()
// boolean	isEmpty()
// Iterator<E>	iterator()
// int	size()
// Object[]	toArray()
// String toString()
